### ISSUE TITLE
Standardize XLA loop APIs

### ISSUE THREAD

**Issue by rpsilva-aws** [pytorch/xla#8918]
## 🚀 Feature

In this item, we standardize the XLA loop APIs that are currently supported with JAX (`scan`, `fori-loop`, `while-loop`), reaching parity with the added UX benefit associated with users wishing to yield the respective benefits.

Scan RFC: https://github.com/pytorch/xla/issues/8620
While-loop/fori-loop RFC: https://github.com/pytorch/xla/issues/6941

There have been diverging variants that can largely benefit in having a single robust implementation, namely:
- Scan: https://github.com/rpsilva-aws/xla/blob/master/torch_xla/experimental/scan.py
- While loop / Fori loop: https://github.com/pytorch/xla/blob/master/torch_xla/experimental/fori_loop.py
- Grad acc (experimental UX): https://github.com/pytorch/xla/blob/master/torch_xla/experimental/gradient_accumulation.py

Given the increasing robustness improvements to `scan` from @tengyifei, we should consider building upon it, and unifying most of the core implementation. A subitem includes replacing grad acc UX with `scan` or `while-loop` entirely, so that customers can use the existing UX to achieve the same benefits as JAX (AXLearn's gradient accumulation, https://github.com/apple/axlearn/pull/614/files).

## Motivation

Currently, the while-loop and fori-loop experimental APIs (https://github.com/pytorch/xla/issues/6941) are limited (issues: https://github.com/pytorch/xla/issues/7839, https://github.com/pytorch/xla/issues/8566), namely robustness, e.g. unable to handle special scalar iteration values (0, 1), preserving the static XLA loop argument requirements (body, condition and init) needed for https://openxla.org/xla/operation_semantics#while, device RNG initialization, among others.

cc: @tengyifei @ManfeiBai
